src/protocol/mod.rs

```diff
@@ -15,8 +15,81 @@ use self::buf::NotEnoughBytesError;
 pub mod buf;
 pub mod types;
 
-/// A string type backed by [`bytes::Bytes`].
-pub type StrBytes = string::String<bytes::Bytes>;
+mod str_bytes {
+    use std::convert::TryFrom;
+    use std::fmt::{Debug, Display, Formatter};
+    use std::ops::Deref;
+    use std::str::Utf8Error;
+    use bytes::Bytes;
+
+    /// A string type backed by [Bytes].
+    #[derive(Clone, Hash, Ord, PartialOrd, PartialEq, Eq, Default)]
+    pub struct StrBytes(Bytes);
+
+    impl StrBytes {
```

Comment:
Wondering if we also want to provide unchecked constructors, although perhaps it's best to proceed with the entirely safe API first and see if anyone has a need for them later.